### PR TITLE
Unify `IterDomain::split` with `SplitTransform` and similar for merge

### PR DESCRIPTION
**Pull request by jacobhinkle**
Currently, `MergeTransform` and `SplitTransform`, which are the classes used to perform reshape operations, do not re-use the splitting and merging operations used in scheduling, namely the static methods `IterDomain::{split,merge}`. This leads to needing to maintain changes in both as I recently found out when debugging #1174. We can actually already notice they have diverged with respect to `IndexType::GatherScatter` handling.

This PR simply calls the static `IterDomain` method from each of the reshape `*Transform` classes. Since those classes also need to set the rfactor-product flag on the produced `IterDomain`s, I added a flag to the static methods that is only used during reshape.

### CODE REVIEW

**Comment by jacobhinkle** [comment]
!build

File: csrc/transform_view.cpp [hunk 1]
```diff
@@ -277,23 +271,13 @@ class SplitTransform final : public ViewTransform {
...
-            .build();
-
-    IrBuilder::create<Split>(factor_id, remainder_id, id, factor, false);
+    auto [factor_id, remainder_id] = IterDomain::split(
+        id,
+        factor,
+        /*inner_split*/ false,
+        /*start_offset*/ nullptr,
+        /*stop_offset*/ nullptr,
+        /*rfactor_domain*/ true);
```
Comment (wujingyue) [inline csrc/transform_view.cpp:280]: https://clang.llvm.org/extra/clang-tidy/checks/bugprone/argument-comment.html

`/*param_name=*/` has the advantage of being checked by clang-tidy. I'm not sure `/*param_name*/` has the same effect. Please ignore if it does. 

```suggestion
        /*inner_split=*/false,
        /*start_offset=*/nullptr,
        /*stop_offset=*/nullptr,
        /*rfactor_domain=*/true);
```

File: csrc/ir/internal_base_nodes.h [hunk 2]
```diff
@@ -120,7 +120,10 @@ class IterDomain : public Val {
   static std::vector<IterDomain*> clone(
       const std::vector<IterDomain*>& domains);
 
-  static IterDomain* merge(IterDomain* outer, IterDomain* inner);
+  static IterDomain* merge(
```
Comment (wujingyue) [inline csrc/ir/internal_base_nodes.h:123]:
```suggestion
  // When `rfactor_domain` is true, also set the `is_rfactor_domain_` flag of the result IterDomain. 
  static IterDomain* merge(
```

**Review by wujingyue — APPROVED** [review]

File: csrc/transform_view.cpp (hunk 1 above)
Comment (jacobhinkle) [reply csrc/transform_view.cpp:280]: That's good to know! Thanks.

**Comment by jacobhinkle** [comment]
!build --diff

**Comment by jacobhinkle** [comment]
!build --diff

**Pull request merged**